src/integration_tests.zig

```diff
@@ -556,3 +438,183 @@ test "vortex smoke" {
         .script_path = script_path,
     });
 }
+
+const TmpCluster = struct {
+    const replica_count = 3;
+    // The test uses this hard-coded address, so only one instance can be running at a time.
+    const addresses = "127.0.0.1:7121,127.0.0.1:7122,127.0.0.1:7123";
```

Comment:
🤔 I think Zig could be running upgrade test and reformat in parallel. Not sure if it already does that, but the intended semantics I think is that zig test spawns several worker processes, and then the zig build controlling process instructs workers to run specific tests. So, there's a single thread per process, but multiple processes. 

Let's maybe pass the starting port in? So that different tests use different ports? Or maybe don't do this, as, hopefully, vortex would fix it via unshare or whatnot.

Reply:
I was going to but since it seemed to be passing I just assumed that zig wasn't running these in parallel.

But yeah, my plan is to replace all of this with vortex once vortex supports upgrades+reformat (which I would like to do in the near term).